dandi/upload.py

```diff
@@ -1,36 +1,49 @@
+from collections import defaultdict
 from functools import reduce
 import os.path
 from pathlib import Path
 import re
+import sys
 import time
+from typing import TYPE_CHECKING, Any, Dict, Iterator, List, Optional, Set, Union
 
 import click
 
 from . import lgr
 from .consts import DRAFT, dandiset_identifier_regex, dandiset_metadata_file
 from .exceptions import NotFoundError
-from .files import DandisetMetadataFile, LocalAsset, find_dandi_files
+from .files import DandiFile, DandisetMetadataFile, LocalAsset, find_dandi_files
 from .utils import ensure_datetime, get_instance, pluralize
 
+if TYPE_CHECKING:
+    if sys.version_info[:2] >= (3, 8):
```

Comment:
same here -- simpler
```suggestion
    if sys.version_info >= (3, 8):
```
?